bundles/org.openhab.binding.jablotron/src/main/java/org/openhab/binding/jablotron/internal/handler/JablotronBridgeHandler.java

```diff
@@ -0,0 +1,452 @@
+            if (!response.isStatus()) {
+                logger.debug("Error during sending user code: {}", response.getErrorMessage());
+            }
+            return response;
+        } catch (TimeoutException e) {
+            logger.debug("sendUserCode timeout exception", e);
+        } catch (InterruptedException e) {
+            logger.debug("sendUserCode exception", e);
+        } catch (JsonSyntaxException e) {
+            logger.debug("JSON syntax exception", e);
+        } catch (ExecutionException e) {
+            if (e.getMessage().contains(AUTHENTICATION_CHALLENGE)) {
+                relogin();
+                throw new SecurityException(AUTHENTICATION_CHALLENGE);
+            } else {
+                logger.debug("sendUserCode exception", e);
+            }
+        }
+        return null;
+    }
+
+    protected synchronized @Nullable List<JablotronHistoryDataEvent> sendGetEventHistory(Thing th, String alarm) {
+        String url = JABLOTRON_API_URL + alarm + "/eventHistoryGet.json";
+        String urlParameters = "{\"limit\":1, \"service-id\":" + th.getUID().getId() + "}";
+
+        try {
+            ContentResponse resp = createRequest(url)
+                    .header(HttpHeader.ACCEPT, APPLICATION_JSON)
+                    .content(new StringContentProvider(urlParameters), APPLICATION_JSON)
+                    .send();
+
+            String line = resp.getContentAsString();
+            logger.trace("get event history: {}", line);
+            JablotronGetEventHistoryResponse response = gson.fromJson(line, JablotronGetEventHistoryResponse.class);
+            if (200 != response.getHttpCode()) {
+                logger.debug("Got error while getting history with http code: {}", response.getHttpCode());
+            }
+            return response.getData().getEvents();
+        } catch (TimeoutException e) {
+            logger.debug("Timeout during getting alarm history!", e);
+        } catch (InterruptedException e) {
+            logger.debug("sendGetEventHistory exception", e);
+        } catch (JsonSyntaxException e) {
+            logger.debug("JSON syntax exception", e);
+        } catch (ExecutionException e) {
+            if (e.getMessage().contains(AUTHENTICATION_CHALLENGE)) {
+                relogin();
+            } else {
+                logger.debug("sendGetEventHistory exception", e);
+            }
+        }
+        return null;
+    }
+
+    protected synchronized @Nullable JablotronDataUpdateResponse sendGetStatusRequest(Thing th) {
+        String url = JABLOTRON_API_URL + "dataUpdate.json";
+        String urlParameters = "data=[{ \"filter_data\":[{\"data_type\":\"section\"},{\"data_type\":\"pgm\"},{\"data_type\":\"thermometer\"},{\"data_type\":\"thermostat\"}],\"service_type\":\"" + th.getThingTypeUID().getId() + "\",\"service_id\":" + th.getUID().getId() + ",\"data_group\":\"serviceData\"}]&system=" + SYSTEM;
```

Comment:
The design you have used here is that a thing id represents a identiffier. However, convention is that the id of a thing should not be used, but can freely be defined by the user. The design should be that the serviceId is a parameter of a thing.

Reply:
OK, changed design so the serviceId is a parameter of the thing